micro-benchmarks/DRB170-flush-orig-no.c

```diff
@@ -0,0 +1,116 @@
+/*
+!!!~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~!!!
+!!! Copyright (c) 2017-20, Lawrence Livermore National Security, LLC
+!!! and DataRaceBench project contributors. See the DataRaceBench/COPYRIGHT file for details.
+!!!
+!!! SPDX-License-Identifier: (BSD-3-Clause)
+!!!~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~!!!
+ */
+/**
+ * LU.C: This file is part of kernel of the NAS Parallel Benchmarks 3.0 LU suit.
+ * Tsan and Intel Inspector can not correctly analysis the flush, and report a false postive.
+*/
+
+#include <stdio.h>
+#include <stdbool.h>
+#define ISIZ 12
+#define	OMEGA_DEFAULT	1.2
+
+#if defined(_OPENMP)
+/* for thread synchronization */
+static bool flag[ISIZ/2*2+1];
+#endif /* _OPENMP */
+
+int main(int argh, char* argv[])
+{
+  int i,j,k,m,ist,iend,jst,jend,nx,ny,nz,iglob,xi,jglob,eta,zeta;
+  double tmp,omega;
+  double v[ISIZ][ISIZ/2*2+1][ISIZ/2*2+1][5];
+  omega = OMEGA_DEFAULT;
+  nx = ISIZ;
+  ny = ISIZ;
+  nz = ISIZ;
+  ist = 1;
+  iend = nx-2;
+  jst =1;
+  jend = ny -2;
+  
+/* inintal value */
+
+  #pragma omp for
+  for (i = 0; i < nx; i++) {
+    iglob = i;
+    xi = ( (double)(iglob) ) / ( nx - 1 );
+    for (j = 0; j < ny; j++) {
+      jglob = j;
+      eta = ( (double)(jglob) ) / ( ny - 1 );
+      for (k = 0; k < nz; k++) {
+        zeta = ( (double)(k) ) / ( nz - 1 );
+        for (m = 0; m < 5; m++) {
+          v[i][j][k][m] = xi + eta
+            +  zeta
+            +  xi * xi
+            +  eta * eta
+            +  zeta * zeta
+            +  xi * xi * xi
+            +  eta * eta * eta
+            +  zeta * zeta * zeta
+            +  xi * xi * xi * xi
+            +  eta * eta * eta * eta
+            +  zeta * zeta * zeta * zeta;
+        }
+      }
+    }
+  }
+  for(k = 1; k <= nz - 2; k++){
+
+  
+    #pragma omp for nowait schedule(static)
+    for (i = ist; i <= iend; i++) {
+
+    #if defined(_OPENMP)      
+      if (i != ist) {
+        while (flag[i-1] == 0) {
+          #pragma omp flush(flag);
```

Comment:
`#pragma omp flush(flag)` should not have a delimiter `;`. Else, it won't compile.